### ISSUE TITLE
Colour Consistency Issues when switching themes

### ISSUE THREAD

**Issue by alanjacobmathew** [WordPress/gutenberg#43832]
### Description

When switching b/w themes the custom colour given to certain blocks changes automatically in the post Editor. This was observed for Heading, Para, List. The issues is mainly for background colour for these fields. 

Themes tested: 
TT2
TT1-block
T21
T20
T19 

Test were done only for post editor and not the FSE editor. For all the blocks, text colour was black, and background colour was red.  Few images below how each results were. Below tests were done without GB plugin installed, but even when it's installed there were issues with colour

1: TT2
![image](https://user-images.githubusercontent.com/33965848/188278283-cd4b02ee-5ed6-40b9-8cdc-b71b0965b730.png)

2: TT1 Blocks
![image](https://user-images.githubusercontent.com/33965848/188278301-d455dcef-71f7-4564-9d49-9c6f29d7d4ca.png)

3: TT1
![image](https://user-images.githubusercontent.com/33965848/188278314-57ba0b50-af95-4922-8085-82b632ebe464.png)

4: T20
![image](https://user-images.githubusercontent.com/33965848/188278325-2314e9b0-742a-4887-9006-88339457473a.png)

5: T19
![image](https://user-images.githubusercontent.com/33965848/188278332-1dff3651-9465-4da6-a204-281b8e99d71d.png)

### Step-by-step reproduction instructions

1:Add para block, heading block, list block
2: Add text color and background color. 
3: Changelthemes between TT2, TT1-block, TT1, T20, T19

### Environment info

WP 6.0.2
php: 7.4
Browser Brave 1.43.88(chromium)
Win 10

### Please confirm that you have searched existing issues in the repo.

Yes

### Please confirm that you have tested with all plugins deactivated except Gutenberg.

Yes

**Comment by carolinan** [comment]
Can you provide more detailed, step by step instructions of how you set up the colors and also what your expectations where?

**Comment by alanjacobmathew** [comment]
I never did anything more than just switching themes.

Steps:

- Install T22 and T20 themes.
- Activate T20
- Open post editor
- Add a list block
- insert content with red(FF0000) as text color and background color as black(000000)  
- Switch theme to T22.

Now the background color for the list block in my case became #1A4548. I noticed it happening for `para` and `heading` too.

Now change the background color in T20 to #1a4548, and change theme, the background color becomes black. This doesn't happen to other colors I tested.

I tested the cases with themes from T22 upto T19, all with the background color black. and for each theme the result differs as given in the images above. The test cases data were in the following form
a: Current theme installed
b: Text color: red
c: Background color: black 

The text outside the images were the output results for that specific theme

**Comment by carolinan** [comment]
To confirm, are the colors you are selecting from the theme color palette, the WordPress default color palette, or custom from the color picker?

The theme color palettes are theme specific, the colors are intended to change when you switch themes. These colors are what gives each theme a unique look.

**Comment by alanjacobmathew** [comment]
![image](https://user-images.githubusercontent.com/33965848/188470884-c61685c9-87c3-467b-bdd6-fcd040d3662d.png)

Default color paletter

**Comment by carolinan** [comment]
If you look carefully on the picture, you can see that the dark green color, which is selected, is in the theme palette, not the default.

**Comment by cbirdsong** [comment]
This seems like it's possibly related to the issues discussed in #38998 and #39372?

**Comment by annezazu** [comment]
Agreed with @cbirdsong - I'd prefer we close this out and defer to those.

**Comment by alanjacobmathew** [comment]
I picked the colour from the color picker and not from the theme palette. But after retesting it again, even if the color is picked individually, if it is a part of theme palette, it automatically assigns it to as Theme's primary color. 
Not sure, but it may affect users if it directly converts one Theme's primary to another theme's primary color set.

If I set an entire page in accordance with a similar color code and if I change the theme the entire color structure is changed. It may not affect those who simply write content without any color styling, but for those who make landing pages it might have an impact. 

![Colour Cons](https://user-images.githubusercontent.com/33965848/189299962-13ab32c7-5471-447f-8911-93764fc23529.gif)
